src/gui/sdlmain.cpp

```diff
@@ -1818,6 +1816,49 @@ static void DisplaySplash(uint32_t time_ms)
 	SDL_Delay(time_ms);
 }
 
+/* For some preference values, we can safely remove comment after # character
+ */
+static std::string NormalizeConfValue(const char *val)
+{
+	std::string pref = val;
+	const auto comment_pos = pref.find('#');
+	if (comment_pos != std::string::npos)
+		pref.erase(comment_pos);
+	trim(pref);
+	lowcase(pref);
+	return pref;
+}
+
+static void SetupWindowResolution(const char *val)
+{
```

Comment:
Awesome; nice to see you've lifted this out into a self-contained function!

Suggest that the outcome of this function is that `sdl.desktop.window.<dims>` is populated no matter what. If  `original` is set, then we populate the initial Window dimensions with `splash_image.<dims>` (or some other sane default - 640x480? 800x600?).